src/main/java/com/dougnoel/sentinel/steps/ImageVerificationSteps.java

```diff
@@ -54,48 +59,78 @@ public static void verifyImageNotMatch(String elementName, String assertion) thr
-	 * the new empty space if their sizes are not equal.
-	 * Then runs, and returns, an ImageComparisonResult of the two images.
-	 * 
+	 * Takes an updated screenshot of the current element, or page, for comparison to an earlier expected image.
+	 * <br><br>
+	 * <i>
+	 *     If screenshots are different sizes, both will be enlarged to the largest dimensions found. Expanded space
+	 * will be set to parent background color for web, upper-left-most color for windows applications.
+	 * </i>
+	 *
 	 * @param elementName String the name of the element to capture and compare 
 	 * or any casing of "page" alone to compare the entire page.
 	 * @throws IOException if file creation does not work
 	 * 
 	 * @return ImageComparisonResult the image comparison result.
 	 */
     private static ImageComparisonResult compareImages(String elementName) throws IOException {
-    	String imageFileName = scenario.getName()+ "_" + elementName + "_" + PageManager.getPage().getName() + ".png";
-    	String resultImageFileName = scenario.getName()+ "_" + elementName + "_" + PageManager.getPage().getName() + ".png";
+		String outputFolder = "ImageComparison/" + scenario.getName();
+    	String actualFileName = PageManager.getPage().getName() + "_" + elementName + "_ACTUAL" + ".png";
+		String expectedFileName = PageManager.getPage().getName() + "_" + elementName + "_EXPECTED" + ".png";
+    	String resultImageFileName = PageManager.getPage().getName() + "_" + elementName + "_RESULT" + ".png";
     	File screenshotFile;
     	Color backgroundColor;
+		PageObjectType pageType = PageManager.getCurrentPageObjectType();
+
+		if(elementName.matches("(?i)^\\s*(page|window)\\s*$")){
+			TakesScreenshot pageScreenshotTool =((TakesScreenshot) Driver.getWebDriver());
+			screenshotFile = pageScreenshotTool.getScreenshotAs(OutputType.FILE);
+
+			switch(pageType){
+				case EXECUTABLE:
+					WindowsElement appWindow = new WindowsElement("window", Map.of("xpath", "/*"));
+					backgroundColor = appWindow.getColorAtOffset().getColor();
+					break;
+				case WEBPAGE:
+					Element body = new Element("body", Map.of("xpath", "//body"));
+					backgroundColor = body.getBackgroundColor();
+					break;
+				case UNKNOWN:
+				default:
+					log.warn("Page object type is of the unhandled {} type. Black will be used as background color fallback.", pageType);
+					backgroundColor = Color.BLACK;
+					break;
+			}
+		} else{
+			screenshotFile = getElement(elementName).getScreenshot();
+
+			switch(pageType){
+				case EXECUTABLE:
+					backgroundColor = ((WindowsElement) getElement(elementName)).getColorAtOffset().getColor();
+					break;
+				case WEBPAGE:
+					backgroundColor = getElement(elementName).getBackgroundColor();
+					break;
+				case UNKNOWN:
+				default:
+					log.warn("Page object type is of the unhandled {} type. Black will be used as background color fallback.", pageType);
+					backgroundColor = Color.BLACK;
+					break;
+			}
+		}
```

Comment:
This code breaks encapsulation. A test step shouldn't need to know the implementation details of the objects it is comparing. The objects need a comparison feature we can call. If they don't have that, we should wrap them in our own object(s) and create that comparison. In that way, if we create a thrid screenshot implementation (say for a mobile device) this code doesn't need to change. Also, in theory, when I take a screenshot of a window and a web page rendering the same data I should be able to prove they match. While currently the implementation of the taking of the screenshot is different, the output should be the same.